addon/metrics-adapters/dynamic-tag-manager.js

```diff
@@ -0,0 +1,116 @@
+import Ember from 'ember';
+import canUseDOM from '../utils/can-use-dom';
+import objectTransforms from '../utils/object-transforms';
+import BaseAdapter from './base';
+
+const {
+  assert,
+  get,
+  set
+} = Ember;
+const {
+  compact
+} = objectTransforms;
+
+export default BaseAdapter.extend({
+
+  eventQueue: [],
+
+  toStringExtension() {
+    return 'DynamicTagManager';
+  },
+
+  init  () {
+    const config = get(this, 'config');
+    const { src } = config;
+
+    assert(`[ember-metrics] You must pass a valid \`src\` to the ${this.toString()} adapter`, src);
+    set(this, 'dtmSrc', src);
+
+    set(this, 'dataLayerNameString', config['dataLayerName'] || 'dtmDataLayer');
+    window[get(this, 'dataLayerNameString')] = {};
+
+    if (canUseDOM) {
+      /* jshint ignore:start */
+      let script = document.createElement("script");
+      script.src = get(this, 'dtmSrc');
+      script.async = true;
+      document.getElementsByTagName('head')[0].appendChild(script);
+      /* jshint ignore:end */
+    }
+
+    this.checkForQueue();
+  },
+
+  pushAndCheck(dtmEvent) {
+    this.eventQueue.push(dtmEvent);
+    this.checkForQueue();
+  },
+
+  checkForQueue() {
+
+    if (this.get('eventQueue').length) {
+      if (this.isSatelliteDefined()) {
+        this.sendEvents();
+      } else {
+        Ember.run.later(() => {
+          this.checkForQueue();
+        }, 50);
+      }
+    }
+  },
+
+  sendEvents() {
+    this.get('eventQueue').forEach((event) => {
+      for (var key of Object.keys(event)) {
+        window[get(this, 'dataLayerNameString')][key] = event[key];
+      }
+      window._satellite.track(event['event']);
+    });
+    this.set('eventQueue', []);
+  },
+
+  dtmObject(compactedOptions) {
```

Comment:
I don't understand what this method is for. You're setting the `event` key on the `dtmEvent` object to be `compactedOptions.event`, then taking all the other keys from `compactedOptions` and setting it on `dtmEvent`... 

Doesn't that just mean you want to return the `compactedOptions` object itself? Why is this transform needed?